uuid.go

```diff
@@ -220,6 +243,47 @@ func (u UUID) MarshalText() (text []byte, err error) {
 	return
 }
 
+// UnmarshalJSON implements the json.Unmarshaler interface.
+// It works the same as UnmarshalText except the quotation marks are removed if present.
+func (u *UUID) UnmarshalJSON(data []byte) (err error) {
+	if len(data) < 32 {
+		err = fmt.Errorf("uuid: UUID string too short: %s", data)
+		return
+	}
+
+	t := data[:]
+
+	if bytes.Equal(t[:9], urnPrefix) {
+		t = t[9:]
+	} else if t[0] == '{' || t[0] == '"' {
```

Comment:
As far as I can see, unmarshal will fail for this json:
{"uuid": "{1234-...}"}

Wouldn't it be enough for UnmarshalJSON to do just this:

return u.UnmarshalText(data[1:len(data)-1])

or even just:

return u.UnmarshalText(data[1:])

Since UnmarshalText ignores trailing bytes.